### ISSUE TITLE
Expand RECEIVED_EVENT parsing

### ISSUE THREAD

**Issue by Yimura** [Yimura/YimMenu#70]
**Is your feature request related to a problem? Please describe.**
Currently multiple events aren't parsed.

- EXPLOSION_EVENT
- KICK_VOTES_EVENT
- ....

**Describe the solution you'd like**
Implement protection toggles to completely block these events as well as parse these events to prevent trash from coming through.

**Comment by maybegreat48** [comment]
See https://github.com/citizenfx/fivem/blob/92cbb7abb99312b114c532475455a6526241481f/code/components/citizen-server-impl/src/state/ServerGameState.cpp#L4138

Example:

```c++
	bool scan_explosion_event(rage::datBitBuffer* buffer) {
		uint16_t f186;
		uint16_t f208;
		int ownerNetId;
		uint16_t f214;
		ExplosionTypes explosionType;
		float damageScale;

		float posX;
		float posY;
		float posZ;

		bool f242;
		uint16_t f104;
		float cameraShake;

		bool isAudible;
		bool f189;
		bool isInvisible;
		bool f126;
		bool f241;
		bool f243;

		uint16_t f210;

		float unkX;
		float unkY;
		float unkZ;

		bool f190;
		bool f191;

		uint32_t f164;

		float posX224;
		float posY224;
		float posZ224;

		bool f240;
		uint16_t f218;
		bool f216;
		
		f186 = buffer->Read<uint16_t>(16);
		f208 = buffer->Read<uint16_t>(13);
		ownerNetId = buffer->Read<uint16_t>(13);
		f214 = buffer->Read<uint16_t>(13); // 1604+
		explosionType = (ExplosionTypes)buffer->ReadSigned<int>(8); // 1604+ bit size
		damageScale = buffer->Read<int>(8) / 255.0f;

		posX = buffer->ReadSignedFloat(22, 27648.0f);
		posY = buffer->ReadSignedFloat(22, 27648.0f);
		posZ = buffer->ReadFloat(22, 4416.0f) - 1700.0f;

		f242 = buffer->Read<uint8_t>(1);
		f104 = buffer->Read<uint16_t>(16);
		cameraShake = buffer->Read<int>(8) / 127.0f;

		isAudible = buffer->Read<uint8_t>(1);
		f189 = buffer->Read<uint8_t>(1);
		isInvisible = buffer->Read<uint8_t>(1);
		f126 = buffer->Read<uint8_t>(1);
		f241 = buffer->Read<uint8_t>(1);
		f243 = buffer->Read<uint8_t>(1); // 1604+

		f210 = buffer->Read<uint16_t>(13);

		unkX = buffer->ReadSignedFloat(16, 1.1f);
		unkY = buffer->ReadSignedFloat(16, 1.1f);
		unkZ = buffer->ReadSignedFloat(16, 1.1f);

		f190 = buffer->Read<uint8_t>(1);
		f191 = buffer->Read<uint8_t>(1);

		f164 = buffer->Read<uint32_t>(32);

		if (f242)
		{
			posX224 = buffer->ReadSignedFloat(31, 27648.0f);
			posY224 = buffer->ReadSignedFloat(31, 27648.0f);
			posZ224 = buffer->ReadFloat(31, 4416.0f) - 1700.0f;
		}
		else
		{
			posX224 = 0;
			posY224 = 0;
			posZ224 = 0;
		}

		auto f168 = buffer->Read<uint32_t>(32);		 // >= 1868: f_168
		

		f240 = buffer->Read<uint8_t>(1);
		if (f240)
		{
			f218 = buffer->Read<uint16_t>(16);

			if (f191)
			{
				f216 = buffer->Read<uint8_t>(8);
			}
		}
		buffer->Seek(0);

		if (explosionType == ExplosionTypes::EXPLOSION_DIR_FLAME ||
			explosionType == ExplosionTypes::EXPLOSION_DIR_FLAME_EXPLODE ||
			explosionType == ExplosionTypes::EXPLOSION_DIR_GAS_CANISTER ||
			explosionType == ExplosionTypes::EXPLOSION_DIR_STEAM ||
			explosionType == ExplosionTypes::EXPLOSION_DIR_WATER_HYDRANT ||
			explosionType == ExplosionTypes::EXPLOSION_BIRD_CRAP ||
			damageScale > 5.0f ||
			cameraShake > 1.0f ||
			isInvisible ||
			!isAudible) {
			return false;
		}
		return true;
	}

	bool scan_increment_stat_event(rage::datBitBuffer* buffer) {
		Hash hash = buffer->Read<Hash>(32);
		int amount = buffer->Read<int>(32);
		buffer->Seek(0);

		switch (hash)
		{
		case RAGE_JOAAT("MPPLY_GAME_EXPLOITS"):
		case RAGE_JOAAT("MPPLY_VC_HATE"):
		case RAGE_JOAAT("MPPLY_EXPLOITS"):
		case RAGE_JOAAT("MPPLY_TC_ANNOYINGME"):
		case RAGE_JOAAT("MPPLY_TC_HATE"):
			return false;
		
		default:
			return true;
		}
	}
```

**Comment by Yimura** [comment]
Woah dope, this parses the event exact values, and if anything doesn't match up we can deny the event.

**Comment by BugisoftRSG** [comment]
EXPLOSION_EVENT implemented in #625

**Comment by Yimura** [comment]
Yeah you're right, most of these have been implemented now.

2 digit issue as well.
